packages/date-picker/src/vaadin-date-picker-overlay-content.js

```diff
@@ -64,62 +67,17 @@ class DatePickerOverlayContent extends ControllerMixin(ThemableMixin(DirMixin(Po
           overflow: hidden;
         }
 
-        [part='months'],
-        [part='years'] {
-          height: 100%;
-        }
-
-        [part='months'] {
-          --vaadin-infinite-scroller-item-height: 270px;
-          position: absolute;
-          top: 0;
-          left: 0;
-          right: 0;
-          bottom: 0;
-        }
-
-        #scrollers[desktop] [part='months'] {
+        :host([desktop]) ::slotted([slot='months']) {
```

Comment:
Changed to set `desktop` state attribute on the host, so that it can be used for styling.